### PR TITLE
Add seller item creation

### PR DESCRIPTION
**Pull request by christophermanahan**
- Add Auction module to manage items
- Add SellController module to allow item management routing
- Add ability for seller to create items
- Validate item creation

### CODE REVIEW

File: lib/ebae/accounts/user.ex [hunk 1]
```diff
@@ -3,11 +3,12 @@ defmodule Ebae.Accounts.User do
 
   import Ecto.Changeset
 
-  alias Ebae.Accounts.Credential
+  alias Ebae.{Accounts.Credential, Auction.Item}
 
   schema "users" do
     field :username, :string
     has_one :credential, Credential, on_replace: :update
+    has_many :item, Item
```
Comment (ssunday) [inline lib/ebae/accounts/user.ex]: wouldn't it be `items`?

File: lib/ebae/auction.ex [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+
+      iex> list_items()
+      [%Item{}, ...]
+
+  """
+  def list_items do
+    Repo.all(Item)
+  end
+
+  @doc """
```
Comment (ssunday) [inline lib/ebae/auction.ex:24]: I'm anti-doc, but 🤷‍♀

File: lib/ebae_web/controllers/sell_controller.ex [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+
+  def create(conn, %{"item" => item}) do
+    conn
+    |> validate_and_create(item)
+    |> create_reply(conn)
+  end
+
+  defp validate_and_create(conn, item) do
+    with true <- validate(item),
+         user <- Auth.current_user(conn),
```
Comment (ssunday) [inline lib/ebae_web/controllers/sell_controller.ex:26]: wouldn't you check if  user exists before validating?

File: lib/ebae_web/controllers/session_controller.ex [hunk 4]
```diff
@@ -28,14 +28,14 @@ defmodule EbaeWeb.SessionController do
     conn
     |> put_flash(:info, "Farewell")
     |> Auth.sign_out()
-    |> redirect(to: "/signin")
+    |> redirect(to: Routes.session_path(conn, :new))
```
Comment (ssunday) [inline lib/ebae_web/controllers/session_controller.ex:31]: yay

File: lib/ebae_web/controllers/sell_controller.ex (hunk 3 above)
Comment (ssunday) [reply lib/ebae_web/controllers/sell_controller.ex:26]: Ah, well, the pipe through does this

File: lib/ebae_web/router.ex [hunk 5]
```diff
@@ -36,4 +36,14 @@ defmodule EbaeWeb.Router do
     get "/signup", RegistrationController, :new
     post "/signup", RegistrationController, :create
   end
+
+  scope "/sell", EbaeWeb do
```
Comment (ssunday) [inline lib/ebae_web/router.ex:38]: resources? https://hexdocs.pm/phoenix/routing.html#resources

File: lib/ebae_web/templates/sell/form.html.eex [hunk 6]
```diff
@@ -0,0 +1,23 @@
...
+
+  <%= label f, :name %>
+  <%= text_input f, :name %>
+  <%= error_tag f, :name %>
+
+  <%= label f, :description %>
+  <%= text_input f, :description %>
+  <%= error_tag f, :description %>
+
+  <%= label f, :initial_price %>
```
Comment (ssunday) [inline lib/ebae_web/templates/sell/form.html.eex]: could this be a partial <%= text_field, f, key %> or something

File: priv/repo/seeds.exs [hunk 7]
```diff
@@ -1,8 +1,17 @@
...
+item_attrs = %{
+  available: true,
+  description: "some description",
+  initial_price: "120.5",
+  name: "some name",
+}
+
+{:ok, user} = Accounts.create_user(user_attrs)
+
+Auction.create_item(Map.put(item_attrs, :user_id, user.id))
```
Comment (ssunday) [inline priv/repo/seeds.exs:17]: seeeedds

File: lib/ebae/auction.ex (hunk 2 above)
Comment (christophermanahan) [reply lib/ebae/auction.ex:24]: Yeah, idk how I feel about these docs. I thought it was convention but they are quite noisy.

File: lib/ebae/accounts/user.ex (hunk 1 above)
Comment (christophermanahan) [reply lib/ebae/accounts/user.ex]: Yes!

File: lib/ebae_web/router.ex (hunk 5 above)
Comment (christophermanahan) [reply lib/ebae_web/router.ex:38]: Clean, thanks!

File: lib/ebae_web/templates/sell/form.html.eex (hunk 6 above)
Comment (christophermanahan) [reply lib/ebae_web/templates/sell/form.html.eex]: I like it! Removed a bunch of duplication across the forms.

**Pull request merged**